### ISSUE TITLE
Reassigning the memory free pointer does not cause an error.

### ISSUE THREAD

**Issue by Subway2023** [paritytech/revive#232]
```solidity
contract TestMStore {
    function testFunction() public pure returns (uint256) {
        uint256 result;
        assembly {
            let value := 42
            mstore(0x40, value)
            result := mload(0x40)
        }
        return result;
    }
}
```
solc+evm: execution error
solc(viaIR)+evm: execution is normal
resolc+vm: execution is normal

I believe this difference is due to resolc compiling from Yul. Therefore, this difference needs to be reflected in the [documentation](https://contracts.polkadot.io/differences_to_eth).

**Comment by athei** [comment]
cc @xermicus

**Comment by xermicus** [comment]
Thank you @Subway2023 for pointing this out. While overriding the free memory pointer is asking for trouble and shouldn't be done, there are more slight differences, so it's a good point nonetheless. [This will be reflected in our docs](https://github.com/paritytech/contract-docs/pull/54).
